compiler/rustc_driver_impl/src/args.rs

```diff
@@ -153,3 +153,40 @@ impl fmt::Display for Error {
 }
 
 impl error::Error for Error {}
+
+/// check if suspect is a confusables of any option or arg in rustc or fake_args
```

Comment:
Suggestion [INTENT 1/2]: also, please describe in the doc comment at a higher level what we consider high-value confusables (versus risk of false positives). Something like:

1. C compiler flag lookalikes: this is indeed the `["optimize", "o0", "o1", "o2", "o3", "ofast", "og", "os", "oz"]` examples.
3. Codegen flags: e.g. `["pt-level"]` of `-opt-level`.

Probably move this higher level doc to the `warn_on_confusing_output_filename_flag` itself, cf. [INTENT 2/2].